routes/route.py

```diff
@@ -1,58 +1,291 @@
+        if (request.files.get("dataset")):
+            flash(
+                "User provided datasets are not yet supported. The default will be used instead.")
+
+        # this primarily exits for use in pipelining data to VeRNAl & RNAMigos to avoid additional calls to the server
+        get_graphs = input.get("get_graphs", default=0, type=int)
+
+        sequence = input.get("sequence", type=str)
+        if (not sequence and not input_file):
+            abort(400, description="Did not receive a sequence as an argument or a file.")
+        elif (sequence and input_file):
+            abort(
+                400, description="Received both a sequence and a file input. Are you using the right endpoint?")
+        elif (not sequence):
+            sequence = input_file.name
+
+        secondary_structure = input.get(
+            "secondary_structure", default="", type=str)
+        secondary_structure_infile = input.get(
+            "secondary_structure_infile", default=0, type=int)
+
+        # a secondary structure can be provided via a fasta file. This means a string cannot be provided, so verify that only one was received
+        # moreover, if a fasta file is provided, a secondary structure should not be provided via string
+        if (secondary_structure_infile and secondary_structure != ""):
+            if (input_file):
+                input_file.close()
+            abort(400, description="Indicated that the secondary structure is provided in the fasta file, but also provided a separate structure.")
+        elif ((input_file_type == "fa" or input_file_type == "fasta") and secondary_structure != ""):
+            input_file.close()
+            abort(400, description="Received a secondary structure as a string, but a fasta file as a sequence. Please provide the single sequence as a string.")
+        elif (secondary_structure_infile):
+            secondary_structure = "infile"
+
+        # the default dataset is always used for now
+        dataset = input.get("dataset", default="reliable_dataset", type=str)
+
+        # load all arguments if they were provided or set the default value
+        arguments["t"] = input.get("score_threshold", default=-2.3, type=float)
+        arguments["samplesize"] = input.get(
+            "sample_size", default=20000, type=int)
+        arguments["theta"] = input.get("theta", default=1, type=int)
+        arguments["lambda"] = input.get("lambda", default=0.35, type=float)
+        arguments["w"] = input.get("window_length", default=200, type=int)
+        arguments["s"] = input.get("step_size", default=100, type=int)
+        if (input.get("modules")):
+            arguments["mod"] = eval(input.get("modules"))
+        else:
+            arguments["mod"] = None
+        arguments["constraints"] = input.get(
+            "constraints", default="", type=str)
+        arguments["aln"] = input.get("alignment_provided", default=0, type=int)
+        arguments["verbose"] = BAYESPAIRING_VERBOSE
```

Comment:
is there an easier way to do this, like all the arguments at once instead of one at a time?

Reply:
Cleaned up in follow-up.